baremetal_network_provisioning/ml2/hp_network_provisioning_driver.py

```diff
@@ -161,20 +137,51 @@ def update_port(self, port_dict):
         SDN controller for provision VLAN on switch port where bare metal
         is connected.
         """
-        return
+        LOG.debug("update_port with port dict %(port_dict)s",
+                  {'port_dict': port_dict})
+        port_id = port_dict['port']['id']
+        bind_requested = port_dict['port']['bind_requested']
+        update_dict = {'neutron_port_id': port_id,
+                       'bind_requested': bind_requested}
+        db.update_hp_ironic_swport_map_with_bind_req(self.context,
+                                                     update_dict)
 
-    def delete_port(self, port_id):
+    def delete_port(self, port_dict):
         """delete_port. This call makes the REST request to the external
 
         SDN controller for un provision VLAN for the switch port where
         bare metal is connected.
         """
-        # TODO(Selvakumar) need to implement the REST call logic
-        neutron_port_dict = {'neutron_port_id': port_id}
-        switch_port_id = db.get_hp_ironic_swport_map_by_id(self.context,
-                                                           neutron_port_dict)
-        switch_port_dict = {'id': switch_port_id.switch_port_id}
-        db.delete_hp_switch_port(self.context, switch_port_dict)
+        LOG.debug("delete_port with port dict %(port_dict)s",
+                  {'port_dict': port_dict})
+        port_id = port_dict['port']['id']
+        rec_dict = {'neutron_port_id': port_id}
+        bind_port_dict = port_dict.get('port')
```

Comment:
This bind_port_dict should be dynamically reconstructed from internal tables we maintain in the backend driver.

The interface should carry only the neutron port id.